drivers/timers/rx8010.c

```diff
@@ -0,0 +1,613 @@
+
+  buffer[1] = rtc_bin2bcd(newtm.tm_sec);
+
+  /* Save minutes (0-59) converted to BCD */
+
+  buffer[2] = rtc_bin2bcd(newtm.tm_min);
+
+  /* Save hour (0-23) with 24-hour time indication */
+
+  buffer[3] = rtc_bin2bcd(newtm.tm_hour);
+
+  /* Save the day of the week (1-7) */
+
+  buffer[4] = rtc_bin2bcd(newtm.tm_wday + 1);
+
+  /* Save the day of the month (1-31) */
+
+  buffer[5] = rtc_bin2bcd(newtm.tm_mday);
+
+  /* Save the month (1-12) with century */
+
+  buffer[6] = rtc_bin2bcd(newtm.tm_mon + 1);
+
+  /* Save the year */
+
+  buffer[7] = rtc_bin2bcd(newtm.tm_year - 100);
+
+  /* Setup the I2C message */
+
+  msg[0].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[0].addr      = RX8010_I2C_ADDRESS;
+  msg[0].flags     = 0;
+  msg[0].buffer    = buffer;
+  msg[0].length    = 8;
+
+  /* Read back the seconds register */
+
+  cmd              = RX8010_TIME_SECR;
+
+  msg[1].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[1].addr      = RX8010_I2C_ADDRESS;
+  msg[1].flags     = 0;
+  msg[1].buffer    = &cmd;
+  msg[1].length    = 1;
+
+  msg[2].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[2].addr      = RX8010_I2C_ADDRESS;
+  msg[2].flags     = I2C_M_READ;
+  msg[2].buffer    = &seconds;
+  msg[2].length    = 1;
+
+  /* Perform the transfer.  This transfer will be repeated if the seconds
+   * count rolls over to a smaller value while writing.
+   */
+
+  do
+    {
+      ret = I2C_TRANSFER(g_rx8010.i2c, msg, 3);
+      if (ret < 0)
+        {
+          rtcerr("ERROR: I2C_TRANSFER failed: %d\n", ret);
+          return ret;
+        }
+    }
+  while ((buffer[1] & RX8010_TIME_SEC_BCDMASK) >
+  (seconds & RX8010_TIME_SEC_BCDMASK));
+
+  return OK;
+}
+
+static int rx8010_init()
+{
+  struct i2c_msg_s msg[2];
+  uint8_t buffer[4];
+  int ret;
+
+  /* set reserved register 0x17 with specified value of 0xd8 */
+
+  buffer[0] = RX8010_RESV17;
+  buffer[1] = 0xd8;
+  msg[0].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[0].addr      = RX8010_I2C_ADDRESS;
+  msg[0].flags     = 0;
+  msg[0].buffer    = buffer;
+  msg[0].length    = 2;
+
+  ret = I2C_TRANSFER(g_rx8010.i2c, msg, 1);
+  if (ret < 0)
+    {
+      rtcerr("ERROR: I2C_TRANSFER failed: %d\n", ret);
+      return ret;
+    }
+
+  /* set reserved register 0x30 with specified value of 0x00 */
+
+  buffer[0] = RX8010_RESV30;
+  buffer[1] = 0x00;
+  msg[0].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[0].addr      = RX8010_I2C_ADDRESS;
+  msg[0].flags     = 0;
+  msg[0].buffer    = buffer;
+  msg[0].length    = 2;
+
+  ret = I2C_TRANSFER(g_rx8010.i2c, msg, 1);
+  if (ret < 0)
+    {
+      rtcerr("ERROR: I2C_TRANSFER failed: %d\n", ret);
+      return ret;
+    }
+
+  /* set reserved register 0x31 with specified value of 0x08 */
+
+  buffer[0] = RX8010_RESV31;
+  buffer[1] = 0x08;
+  msg[0].frequency = CONFIG_RX8010SJ_I2C_FREQUENCY;
+  msg[0].addr      = RX8010_I2C_ADDRESS;
+  msg[0].flags     = 0;
+  msg[0].buffer    = buffer;
+  msg[0].length    = 2;
```

Comment:
```suggestion
```
I think most of `msg[0]` assignments can be done once. Only `msg[0].length` can be corrected before `I2C_TRANSFER`.

Does RX8010 support multi register access so all config can be written with a single I2C transaction?